src/ansys/tools/repo_sync/repo_sync.py

```diff
@@ -1,214 +1,141 @@
+    repo_path = os.path.join(temp_dir.name, repository)
+    authenticated_url = f"https://{token}@{pygithub_repo.html_url.split('https://')[-1]}"
+    Repo.clone_from(authenticated_url, repo_path)
+
+    # Copy local folder contents to the cloned repository
+    destination_path = os.path.join(repo_path, to_dir)
+    print(f">>> Moving desired files from {from_dir} to {destination_path} ...")
+    os.makedirs(destination_path, exist_ok=True)
+    shutil.copytree(
+        from_dir,
+        os.path.join(destination_path),
+        ignore=shutil.ignore_patterns(*prohibited_extensions),
+        dirs_exist_ok=True,
+    )
+
+    # Commit changes to a new branch
+    print(f">>> Checking out new branch '{new_branch_name}' from '{branch_checked_out}'...")
+    repo = Repo(repo_path)
+    try:
+        repo.git.checkout(branch_checked_out)
+        repo.git.checkout("-b", new_branch_name)
+        print(f">>> Committing changes to branch '{new_branch_name}'...")
+        repo.git.add("--all")
+        repo.index.commit("sync: add changes from local folder")
+
+        pull_request = None
+        if not dry_run:
+            # Push changes to remote repositories
+            print(f">>> Force-pushing branch '{new_branch_name}' remotely...")
+            repo.git.push("--force", "origin", new_branch_name)
+
+            # Create a pull request
+            try:
+                print(f">>> Creating pull request from '{new_branch_name}'...")
+                pull_request = pygithub_repo.create_pull(
+                    title=pr_title,
+                    body="Please review and merge these changes.",
+                    base=branch_checked_out,
+                    head=new_branch_name,
+                )
+            except GithubException as err:
+                if err.args[0] == 422 or err.data["message"] == "Validation Failed":
+                    print(f">>> Branch and pull request already existed, searching for it...")
+
+                    # Pull request already exists
+                    prs = pygithub_repo.get_pulls()
+
+                    # Find the associated PR (must be opened...)
+
+                    associated_pull_request = None
+                    for pr in prs:
+                        if pr.head.ref == new_branch_name:
+                            associated_pull_request = pr
+                            break
+
+                    # Return the associated PR
+                    if associated_pull_request:
+                        pull_request = associated_pull_request
+                    else:
+                        # Don't know what could have happened...
+                        raise err
+                else:
+                    # Unidentified error.. raise it.
+                    raise err
+
+            print(f">>> Pull request created: {pull_request.html_url}")
+            return pull_request.html_url
+        else:
+            print(f">>> Dry run successful.")

```

Comment:
For the dry run option, could we list the name of the files modified? Could be useful.

Reply:
We could in fact do it for all cases. It would also be useful for the nominal case scenario